app/controllers/spree/api/favorites_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Spree
+  module Api
+    class FavoritesController < Spree::Api::BaseController
+
+      def index
+        @favorites = @current_api_user.favorites
+                                      .page(params[:page])
+                                      .per(params[:per_page])
+      end
+
+      def create
+        @favorite = @current_api_user.favorites
+                                     .new(favorable_id: params[:product_id],
+                                          favorable_type: "Spree::Product")
+
+        @success = @favorite.save
+        status = @success ? 200 : 422
+
+        render status: status
+      end
+
+      def unfavorite
+        @favorite = @current_api_user.favorites
+                                     .find_by(favorable_id: params[:product_id],
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.